dbms/src/Interpreters/Aggregator.cpp

```diff
@@ -645,9 +692,19 @@ ALWAYS_INLINE void Aggregator::executeImplBatch(
     {
         /// For all rows.
         AggregateDataPtr place = aggregates_pool->alloc(0);
-        for (size_t i = agg_process_info.start_row; i < agg_process_info.start_row + agg_size; ++i)
-            state.emplaceKey(method.data, i, *aggregates_pool, sort_key_containers).setMapped(place);
-        agg_process_info.start_row += agg_size;
+        try
+        {
+            for (size_t i = 0; i < agg_size; ++i)
+            {
+                state.emplaceKey(method.data, agg_process_info.start_row, *aggregates_pool, sort_key_containers)
+                    .setMapped(place);
+                ++agg_process_info.start_row;
+            }
+        }
+        catch (ResizeException &)
+        {
+            LOG_INFO(log, "HashTable resize throw ResizeException since the data is already marked for spill");
+        }
```

Comment:
```suggestion
            for (size_t i = 0; i < agg_size; ++i)
            {
                auto res = emplaceKey(state, method.data, agg_process_info.start_row, *aggregates_pool, sort_key_containers)
                    .setMapped(place);
                if (!state)
                {
                    LOG_INFO(log, "HashTable resize throw ResizeException since the data is already marked for spill");
                    break;
                }
                ++agg_process_info.start_row;
            }
```

Reply:
Done